pkg/authn/kubernetes/keychain.go

```diff
@@ -58,6 +58,12 @@ type Options struct {
 	// ImagePullSecrets holds the names of the Kubernetes secrets (scoped to
 	// Namespace) containing credential data to use for the image pull.
 	ImagePullSecrets []string
+
+	// UseMountSecrets determines whether or not mount secrets in the ServiceAccount
+	// should be considered. Mount secrets are those listed under the `.secrets`
+	// attribute of the ServiceAccount resource. Ignored if ServiceAccountName is set
+	// to NoServiceAccount.
+	UseMountSecrets bool
```

Comment:
I decided to be overly cautious by introducing a flag for the new behavior. Happy to do away with it if the maintainers deem it unnecessary.